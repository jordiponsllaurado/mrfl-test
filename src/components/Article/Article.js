import PropTypes from 'prop-types';

const Article = ({ title, description, image }) => {
  return (
    <div className="article">
      {image && (
        <div className="article-image">
          <img src={image.imageUrl} alt={image.altValue} />
        </div>
      )}
      <div className="article-content">
        <div className="article-title">{title}</div>
        <div className="article-description">{description}</div>
      </div>
    </div>
  );
};

Article.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.shape({
    imageUrl: PropTypes.string,
    altValue: PropTypes.string
  })
};

Article.defaultProps = {
  image: null
};

export default Article;
