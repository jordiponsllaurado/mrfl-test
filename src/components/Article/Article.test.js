import { render } from '@testing-library/react';
import Article from './Article';

test('renders Article with an image', () => {
  const image = {
    imageUrl: 'https://picsum.photos/200/300',
    altValue: 'Google'
  };
  const { container } = render(
    <Article title="Article title" description="Article description" image={image} />
  );
  expect(container).toMatchSnapshot();
});

test('renders Article without an image', () => {
  const { container } = render(<Article title="Article title" description="Article description" />);
  expect(container).toMatchSnapshot();
});
