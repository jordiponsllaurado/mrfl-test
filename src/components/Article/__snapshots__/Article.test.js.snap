// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Article with an image 1`] = `
<div>
  <div
    class="article"
  >
    <div
      class="article-image"
    >
      <img
        alt="Google"
        src="https://picsum.photos/200/300"
      />
    </div>
    <div
      class="article-content"
    >
      <div
        class="article-title"
      >
        Article title
      </div>
      <div
        class="article-description"
      >
        Article description
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Article without an image 1`] = `
<div>
  <div
    class="article"
  >
    <div
      class="article-content"
    >
      <div
        class="article-title"
      >
        Article title
      </div>
      <div
        class="article-description"
      >
        Article description
      </div>
    </div>
  </div>
</div>
`;
